sources/System/Renderer/RenderSystem.cpp

```diff
@@ -31,16 +33,34 @@ namespace BBM
 		this->_window.clear();
 
 		this->_window.useCamera(this->_camera);
-		for (auto &[_, pos, drawable] : this->_wal.scene->view<PositionComponent, Drawable3DComponent>()) {
+		for (auto &[entity, pos, drawable] : this->_wal.scene->view<PositionComponent, Drawable3DComponent>()) {
+			auto *modelShader = entity.tryGetComponent<ShaderComponentModel>();
+
+			if (modelShader) {
+				auto &model = dynamic_cast<RAY::Drawables::Drawables3D::Model &>(*drawable.drawable);
```

Comment:
Please store the dynamic casted on a variable, dynamic_cast is costly. Don't do it twice per frame.